libraries/aws/provisioning/test/system/aws_iot_tests_provisioning_system.c

```diff
@@ -428,11 +536,58 @@ TEST( Provisioning_System, CreateKeysAndCertificateNominalCase )
 }
 
 /**
- * @brief Tests the behavior of the Provisioning RegisterThing API in the nominal (or success) case where the server
- * responds
- * within the specified timeout period.
+ * @brief Verifies the behavior of the CreateCertificateFromCsr API in making
+ * a request for certificate-creation from CSR with the server.
  */
-TEST( Provisioning_System, RegisterThingNominalCase )
+TEST( Provisioning_System, CreateCertFromCsrNominalCase )
+{
+    AwsIotProvisioningError_t status = AWS_IOT_PROVISIONING_SUCCESS;
+    AwsIotProvisioningCreateCertFromCsrCallbackInfo_t callbackInfo =
+    {
+        .userParam = NULL,
+        .function  = _printCertificateFromCsrCallback
+    };
+
+    status = AwsIotProvisioning_CreateCertificateFromCsr( _mqttConnection,
+                                                          IOT_MQTT_QOS_1,
+                                                          AWS_IOT_TEST_PROVISIONING_CSR_PEM,
+                                                          sizeof( AWS_IOT_TEST_PROVISIONING_CSR_PEM ),
+                                                          AWS_IOT_TEST_PROVISIONING_TIMEOUT,
+                                                          &callbackInfo );
+
+    TEST_ASSERT_EQUAL( AWS_IOT_PROVISIONING_SUCCESS, status );
+}
+
+/**
+ * @brief Verifies that the CreateCertificateFromCsr API returns an error and correctly
+ * parses a rejected server response server when called with an invalid CSR string.
+ */
+TEST( Provisioning_System, CreateCertFromCsrWithInvalidCsr )
+{
+    AwsIotProvisioningError_t status = AWS_IOT_PROVISIONING_SUCCESS;
+    AwsIotProvisioningCreateCertFromCsrCallbackInfo_t callbackInfo =
+    {
+        .userParam = NULL,
+        .function  = _checkInvalidCsrServerResponseCallback
+    };
+
+    const char * pInvalidCsr = "GibberishCsr";
+
+    status = AwsIotProvisioning_CreateCertificateFromCsr( _mqttConnection,
+                                                          IOT_MQTT_QOS_1,
+                                                          pInvalidCsr,
+                                                          strlen( pInvalidCsr ),
+                                                          AWS_IOT_TEST_PROVISIONING_TIMEOUT,
+                                                          &callbackInfo );
+
+    TEST_ASSERT_EQUAL( AWS_IOT_PROVISIONING_SERVER_REFUSED, status );
+}
+
+/**
+ * @brief Tests the behavior of the RegisterThing API function in the workflow when
+ * keys are generated by generated through the CreateKeysAndCertificate API
```

Comment:
"when keys are generated by generated through" sounds like something was missed in editing?

Reply:
Fixed